modules/tensor_mechanics/src/materials/GeneralizedKelvinVoigtBase.C

```diff
@@ -0,0 +1,138 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "GeneralizedKelvinVoigtBase.h"
+
+template <>
+InputParameters
+validParams<GeneralizedKelvinVoigtBase>()
+{
+  InputParameters params = validParams<LinearViscoelasticityBase>();
+  return params;
+}
+
+GeneralizedKelvinVoigtBase::GeneralizedKelvinVoigtBase(const InputParameters & parameters)
+  : LinearViscoelasticityBase(parameters)
+{
+}
+
+void
+GeneralizedKelvinVoigtBase::computeQpApparentElasticityTensors()
+{
+  _instantaneous_elasticity_tensor[_qp] = _first_elasticity_tensor[_qp];
+  _apparent_elasticity_tensor[_qp] = _first_elasticity_tensor[_qp].invSymm();
+
+  for (unsigned int i = 0; i < _springs_elasticity_tensors[_qp].size(); ++i)
+  {
+    double theta_i = computeTheta(_dt, _dashpot_viscosities[_qp][i]);
+    double gamma = _dashpot_viscosities[_qp][i] / (_dt * theta_i);
+    _apparent_elasticity_tensor[_qp] +=
+        _springs_elasticity_tensors[_qp][i].invSymm() / (1. + gamma);
```

Comment:
These invSymm() operations are going to be expensive, yes?  I wonder whether it's worth thinking of coding some sort of shortcut would be good.  A few weeks ago i wrote a PR (#9323) in which the compliance tensor (inverse of E) was computed in E's constructor, and then classes could use that rather than doing invSymm() all the time.  Perhaps we could resurrect that idea here?  Some sort of profiling might be useful.